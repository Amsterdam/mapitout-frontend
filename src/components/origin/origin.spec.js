import { shallowMount, createLocalVue } from "@vue/test-utils";

import ComponentOrigin from "./origin.vue";
import { mergeDeepLeft } from "ramda";

const localVue = createLocalVue();

const createWrapper = (options = {}) =>
  shallowMount(
    ComponentOrigin,
    mergeDeepLeft(options, {
      localVue,
      stubs: ["component-address-type", "component-input-address"],
      propsData: {
        value: {
          typeId: 0,
          addressId: "test",
          address: "test",
          addressLat: 0,
          addressLng: 0
        },
        types: [],
        searchAddress: jest.fn(),
        geolocateAddress: jest.fn()
      }
    })
  );

describe("ComponentOrigin", () => {
  it("should create", () => {
    const wrapper = createWrapper();

    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it("should emit an input event when the address type changes its value", () => {
    const wrapper = createWrapper();

    wrapper.setData({
      typeId: 2
    });

    expect(wrapper.emitted().input).toBeTruthy();
    expect(wrapper.emitted().input.length).toBe(1);
    expect(wrapper.emitted().input[0]).toMatchSnapshot();
  });

  it("should emit and input event when the address changes its value", () => {
    const wrapper = createWrapper();

    wrapper.vm.$data.address = {
      id: "different",
      address: "different",
      lat: 5,
      lng: 4
    };

    expect(wrapper.emitted().input).toBeTruthy();
    expect(wrapper.emitted().input.length).toBe(1);
    expect(wrapper.emitted().input[0]).toMatchSnapshot();
  });
});
