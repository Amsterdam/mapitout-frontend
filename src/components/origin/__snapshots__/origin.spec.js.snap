// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentOrigin should emit an input event when the address type changes its value 1`] = `
Array [
  Object {
    "address": "test",
    "addressId": "test",
    "addressLat": 0,
    "addressLng": 0,
    "typeId": 2,
  },
]
`;

exports[`ComponentOrigin should emit and input event when the address changes its value 1`] = `
Array [
  Object {
    "address": "different",
    "addressId": "different",
    "addressLat": 5,
    "addressLng": 4,
    "typeId": 0,
  },
]
`;
